src/components/data_display/helpers.ts

```diff
@@ -203,14 +243,78 @@ const buildColumnDisplayDefinitions = async (
   return columnDisplayDefinitions
 }
 
+const dataTypeFilterListMap = {
+  string: true,
+  number: false,
+  boolean: true,
+  Date: false,
+}
+
+const buildFilterDefinitions = (
+  filterColumns: string[],
+  columnDefinitions: ColumnDisplayDefinitions,
+  fieldDataTypes: { [key: string]: string }
+): FilterDefinition[] =>
+  filterColumns.map((column) => {
+    const customDefinition = columnDefinitions[column]
+    const dataType = fieldDataTypes[column] as keyof typeof dataTypeFilterListMap
+
+    const defaultShowFilterList = dataTypeFilterListMap?.[dataType] ?? false
+
+    if (!customDefinition)
+      // "Standard" column with no special filter specifications
+      return {
+        column,
+        title: startCase(column),
+        dataType,
+        showFilterList: defaultShowFilterList,
+        searchFields: [column],
+      }
+
+    if (!customDefinition.filterParameters)
+      return {
+        column,
+        title: customDefinition.title ?? startCase(column),
+        dataType,
+        showFilterList: defaultShowFilterList,
+        searchFields: [column],
+      }
+
+    const {
+      title,
+      filterParameters: {
+        searchFields,
+        showFilterList,
+        delimiter,
+        valueMap,
+        booleanMapping,
+        ...other
+      },
+    } = columnDefinitions[column]
+
+    return {
+      column,
+      title: title ?? startCase(column),
+      dataType,
+      showFilterList: showFilterList ?? defaultShowFilterList,
+      searchFields: searchFields ?? [column],
+      delimiter,
+      booleanMapping: valueMap ?? booleanMapping,
```

Comment:
`booleanMapping` and `valueMap` are the same thing. I prefer the term `valueMap`, but the front-end is already using `booleanMapping` for Boolean filters, so we return it to match the front-end to avoid additional renaming there. But I'd rather define it in configurations as `valueMap`, so I've allowed both names for the same thing.